source/simulator/core.cc

```diff
@@ -707,15 +705,13 @@ namespace aspect
         for (unsigned int c=0; c<introspection.n_compositional_fields; ++c)
           for (const auto p : boundary_composition_manager.get_fixed_composition_boundary_indicators())
             {
-              auto lambda = [&] (const Point<dim> &x) -> double
+              VectorFunctionFromScalarFunctionObject<dim> vector_function_object(
+                [&] (const Point<dim> &x) -> double
               {
                 return boundary_composition_manager.boundary_composition(p, x, c);
-              };
-
-              VectorFunctionFromScalarFunctionObject<dim> vector_function_object(
-                lambda,
-                introspection.component_masks.compositional_fields[c].first_selected_component(),
-                introspection.n_components);
+              },
+              introspection.component_masks.compositional_fields[c].first_selected_component(),
```

Comment:
I don't mind passing the lambda directly to the constructor, but this confuses ``astyle``, sadly. Do you mind that the indentation is bad? (I think this is why we originally named the lambda)

Reply:
I can live with the indentation. How do others feel?